### DISCUSSION TITLE
Change Animation timing

### DISCUSSION THREAD

**Discussion by waheedakhtar694** [software-mansion/react-native-screens#1112] [Q&A] [answered]
Is there a way to change the animation timing while navigating ? I explore the documentation but didn't find anything in it.

**Comment by kacperkapusciak** [comment]
Hi @waheedakhtar694,
Currently, animation customization is not supported in `native-stack`. You can use fully-customizable [Stack](https://reactnavigation.org/docs/stack-navigator) implemented in JS for that.

**Comment by waheedakhtar694** [comment]
@kacperkapusciak I was using JS stack but due to performance issues specifically on android i moved to native stack and it works better on android but the animation while navigation is specifically sliding from right or left is slow so i wants to make a little faster, any plan to add this support in the near future ?

**Comment by WoLewicki** [comment, marked as answer]
If you need to change the duration of animation, you can change it globally in `res` files of `react-native-screens`, e.g. for `slide_from_right` you need changes in duration here: https://github.com/software-mansion/react-native-screens/blob/afd0b560a480377cec98fd43c9b2a2a284ce8c03/android/src/main/res/anim/rns_slide_in_from_right.xml#L3 and here: https://github.com/software-mansion/react-native-screens/blob/master/android/src/main/res/anim/rns_slide_out_to_left.xml#L3 (you can pass a number in that field). Remember to always change both parts of the animation, they can be seen here for `slide_from_right`: https://github.com/software-mansion/react-native-screens/blob/master/android/src/main/java/com/swmansion/rnscreens/ScreenStack.kt#L194.

**Reply by waheedakhtar694** [reply]
Thanks @WoLewicki it works.
